napari/settings/_yaml.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from pydantic import BaseModel
+from yaml import Dumper, SafeDumper, dump_all
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping
+    from typing import AbstractSet, Any, Dict, Optional, TypeVar, Union
+
+    IntStr = Union[int, str]
+    AbstractSetIntStr = AbstractSet[IntStr]
+    DictStrAny = Dict[str, Any]
+    MappingIntStrAny = Mapping[IntStr, Any]
+    Model = TypeVar('Model', bound=BaseModel)
+
+
+class YamlDumper(SafeDumper):
+    """The default YAML serializer for our pydantic models.
+
+    Add support for custom types by using `YamlDumper.add_representer` or
+    `YamlDumper.add_multi_representer` below.
+    """
+
+
+# add_representer requires a strict type match
+# add_multi_representer also works for all subclasses of the provided type.
+YamlDumper.add_multi_representer(str, YamlDumper.represent_str)
+YamlDumper.add_multi_representer(
+    Enum, lambda dumper, data: dumper.represent_str(data.value)
+)
+YamlDumper.add_representer(
+    set, lambda dumper, data: dumper.represent_list(data)
+)
+YamlDumper.add_multi_representer(
+    set, lambda dumper, data: dumper.represent_list(data)
+)
```

Comment:
# add_representer requires a strict type match
# add_multi_representer also works for all subclasses of the provided type.

If so,  do you need `add_representer` for set ? Wouldn't the second one also take care of the set; or do multi only work for subclases

Reply:
I tried that first, but if a single representer exists it will take priority over the multi_representer... and in this case, the default Dumper has a set representer that i want to override, so it was necessary (what's not really necessary is the multi representer, that was more for the future... maybe I'll remove that until we need it).  Here's the default representer:
```yaml
plugins:
  disabled_plugins: !!set
    bioformats: null
```
but I just want it to be a list (pydantic will turn it into a set because of the schema)
```yaml
plugins:
  disabled_plugins:
  - bioformats
```